src/Numerics/Mesh/Filters.jl

```diff
@@ -212,7 +212,7 @@ function χ(x)
     if (x == 0.5)
```

Comment:
Should this really be exact equality? Also, some floating point instability issue (since `0.5` is a `Float64` and `x` isn't necessarily)

Reply:
The equality should be exact since 0.5 generates a division by zero in the else statement. I'll adjust it to deal with the floating point instability issue, thanks for pointing this out.

Reply:
Still not sure this is right then.

Maybe:

```suggestion
    if (2x -1 == 0)
```

@simonbyrne?

Really there should probably be an asymptotic expansion when `|2x-1|` is small since even with small values you can get issues (I would think)

Reply:
This seems plausible, I'll differ to your expertise on this.

Reply:
🤷

Reply:
Yeah, this does look like it will suffer from cancellation error: using `log1p` might help a bit, but the division might cause problems.

Reply:
The division is fine, since `2abs(η) - 1` has to be at least `eps()/2`: it was the cancellation in the `log(1+...)` which was the problem.